src/storage/raw/raw_mvcc.rs

```diff
@@ -238,3 +238,127 @@ impl<I: Iterator> Iterator for RawMvccIterator<I> {
             .unwrap_or_else(|| self.inner.value())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::storage::raw::encoded::RawEncodeSnapshot;
+    use crate::storage::TestEngineBuilder;
+    use api_version::{APIVersion, RawValue, APIV2};
+    use engine_traits::raw_ttl::ttl_to_expire_ts;
+    use engine_traits::CF_DEFAULT;
+    use kvproto::kvrpcpb::{ApiVersion, Context};
+    use std::fmt::Debug;
+    use std::iter::Iterator as StdIterator;
+    use std::sync::mpsc::{channel, Sender};
+    use tikv_kv::{Engine, Iterator as EngineIterator, Modify, WriteData};
+
+    fn expect_ok_callback<T: Debug>(done: Sender<i32>, id: i32) -> tikv_kv::Callback<T> {
+        Box::new(move |x: tikv_kv::Result<T>| {
+            x.unwrap();
+            done.send(id).unwrap();
+        })
+    }
+
+    #[test]
+    fn test_raw_mvcc_snapshot() {
+        let engine = TestEngineBuilder::new()
+            .api_version(ApiVersion::V2)
+            .build()
+            .unwrap();
+
+        let (tx, rx) = channel();
+        let ctx = Context {
+            api_version: ApiVersion::V2,
+            ..Default::default()
+        };
+
+        let test_data = vec![
+            (b"r\0a".to_vec(), b"aa".to_vec(), 10),
```

Comment:
The `r` prefix should be represented by a const.

Reply:
I mark this issue as `TODO` (see [here](https://github.com/tikv/tikv/pull/12343/files#diff-453305dc4c5ec74aa300e9e75899dfbf824382807c887451bf2155b4d505aa38R276)), to avoid the PR being too large and It will be fixed later.